modules/javafx.base/src/main/java/javafx/beans/binding/ListExpression.java

```diff
@@ -235,7 +235,7 @@ public boolean contains(Object obj) {
     @Override
     public Iterator<E> iterator() {
         final ObservableList<E> list = get();
-        return (list == null)? EMPTY_LIST.iterator() : list.iterator();
+        return (list == null)? ListExpression.<E>emptyList().iterator() : list.iterator();
```

Comment:
You're using three slightly different ways of referring to the empty list:
* `ListExpression.<E>emptyList()`
* `emptyList()`
* `EMPTY_LIST`

What do you think about using the first option in all cases?

Reply:
I'm fine with that; the first two are equivalent, but in some cases I need to add the type witness to avoid a warning and you can only do that by adding the class name as well (ie. `<E>emptyList()` is not allowed, but `ListExpression.<E>emptyList()` is.

Reply:
I've removed the repeated checks from these classes.